src/lualib/ArrayReduce.ts

```diff
@@ -1,20 +1,23 @@
+/** @vararg */
+interface VarArg<T> extends Array<T> {}
+
 // https://www.ecma-international.org/ecma-262/9.0/index.html#sec-array.prototype.reduce
 function __TS__ArrayReduce<T>(
     this: void,
     arr: T[],
     callbackFn: (accumulator: T, currentValue: T, index: number, array: T[]) => T,
-    initial?: T
+    ...initial: VarArg<T>
 ): T {
     const len = arr.length;
 
-    if (len === 0 && initial === undefined) {
+    if (len === 0 && select("#", ...initial) === 0) {
```

Comment:
That's out of scope of this PR, but we probably want to transform `vararg.length` to `select("#", ...)` in general